graph-subscriptions-api/src/schema.rs

```diff
@@ -118,120 +123,88 @@ impl RequestTicketDto {
-            .iter()
-            .cloned()
-            .collect::<HashSet<_>>();
-
-        Ok(unq_subgraph_deployment_qm_hashes.len() as u32)
+    async fn queried_subgraphs_count(&self) -> i64 {
+        self.queried_subgraphs_count
     }
     /// List of `Subgraph` records that this request ticket queried.
     async fn queried_subgraphs<'ctx>(
         &self,
         ctx: &Context<'ctx>,
-        _first: Option<i32>,
-        _skip: Option<i32>,
+        first: Option<i32>,
+        skip: Option<i32>,
     ) -> Result<Option<Vec<Subgraph>>> {
+        let first = first.unwrap_or(100);
+        let skip = skip.unwrap_or(0);
+        let ticket_payload_wrapper = ctx.data_opt::<TicketPayloadWrapper>();
+        if ticket_payload_wrapper.is_none() {
+            return Err(AuthError::Unauthorized.into());
+        }
+        let ticket_payload = ticket_payload_wrapper.unwrap();
+        let payload = &ticket_payload.ticket_payload;
+        let user = Address(payload.user.unwrap_or(payload.signer).0);
         let schema_ctx = ctx
             .data_unchecked::<Arc<Mutex<GraphSubscriptionsSchemaCtx>>>()
             .lock()
             .await;
-        let mut conn = schema_ctx
+        let uniq_deployment_hashes: Vec<DeploymentId> = schema_ctx
             .datasource
-            .redis_client
-            .get_async_connection()
-            .await?;
-        let path = format!(
-            "'$..{}[?(@.ticket_name==\"{}\")].deployment_qm_hash'",
-            self.ticket_user.to_string().to_lowercase(),
-            self.ticket_name.to_string()
-        );
-        let subgraph_deployment_qm_hashes: Vec<String> = conn
-            .json_get(
+            .uniq_deployments_for_ticket(user, self.ticket_name.to_string())
+            .await?
+            .into_iter()
+            .map(|d| d.deployment_qm_hash)
+            .collect();
+        let mut subgraphs_futures = FuturesUnordered::new();
+        // iterate through deployment hashes, get the Subgraphs associated to each
+        for deployment_hash in uniq_deployment_hashes {
+            let deployment = deployment_hash.clone();
+            subgraphs_futures.push(
                 schema_ctx
-                    .datasource
-                    .graph_subscriptions_query_result_key
-                    .to_string(),
-                path,
-            )
-            .await?;
-        if subgraph_deployment_qm_hashes.is_empty() {
-            return Ok(Some(vec![]));
+                    .subgraph_deployments
+                    .deployment_subgraphs_res(deployment),
+            );
+        }
+        // flat map all subgraphs and return
+        let mut subgraphs = Vec::<Subgraph>::new();
+        while let Some(subgraph_result) = TryStreamExt::try_next(&mut subgraphs_futures)
+            .await
+            .unwrap()
+        {
+            for found_subgraph in subgraph_result.to_vec() {
+                subgraphs.push(found_subgraph);
+            }
         }
```

Comment:
(referenced in a comment above)
This is also probably a hack due to my lack of rust knowledge/experience. Ideally, this would use `flat_map` on the `uniq_deployment_hashes` vec retrieved from the database. However, I could not for the life of me (no matter how much googling/chat GPTing I did) figure out how to iterate through the deployment qm hashes, and use the `schema_ctx.subgraph_deployments.deployment_subgraphs(&deployment)` to get a `Vec<Subgraph>` from the `Vec<DeploymentId>`. Would love some eyes on this cc @Theodus on a better way?? or maybe this is acceptable. It does work, I tested that.

Reply:
Gahhh. Thank you so much. Much cleaner. Thank you